File: LitterBox.Redis/RedisConfiguration.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="RedisConfiguration.cs" company="SyndicatedLife">
//   Copyright(c) 2017 Ryan Wilson &amp;lt;syndicated.life@gmail.com&amp;gt; (http://syndicated.life/)
//   Licensed under the MIT license. See LICENSE.md in the solution root for full license information.
// </copyright>
// <summary>
//   RedisConfiguration.cs Implementation
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace LitterBox.Redis {
    using LitterBox.Models;

    /// <summary>
    ///     Configuration For Connection
    /// </summary>
    public class RedisConfiguration : BaseConnectionConfiguration {
        /// <summary>
        ///     Database
        /// </summary>
        public int DatabaseID { get; set; } = 0;

        /// <summary>
        ///     Hostname
        /// </summary>
        public string Host { get; set; } = "127.0.0.1";

        /// <summary>
        ///     Password
        /// </summary>
        public string Password { get; set; } = string.Empty;

        /// <summary>
        ///     Port
        /// </summary>
        public int Port { get; set; } = 6380;

        /// <summary>
        ///     UseSSL On Connection
        /// </summary>
        public bool UseSSL { get; set; } = true;
    }
}

File: node/packages/LitterBox.Redis/src/RedisConfiguration.js
// @flow

import { BaseConnectionConfiguration } from '@icehunter/litterbox';

export type RedisConfigurationProps = {
  DataBaseID?: number,
  Host?: string,
  Password?: string,
  Port?: number
};

export class RedisConfiguration extends BaseConnectionConfiguration {
  constructor(props: RedisConfigurationProps = {}) {
    super(props);
    this.DataBaseID = props.DataBaseID || 0;
    this.Host = props.Host || '127.0.0.1';
    this.Password = props.Password || '';
    this.Port = props.Port || 6380;
  }
  DataBaseID: number;
  Host: string;
  Password: string;
  Port: number;
}
